components/admin_console/team_channel_settings/channel/details/channel_moderation.tsx

```diff
@@ -0,0 +1,347 @@
+        },
+
+        disabled_guests_due_to_create_posts: {
+            id: 'admin.channel_settings.channel_moderation.channelMentions.disabledGuestsDueToCreatePosts',
+            defaultMessage: 'Guests can not use channel mentions without the ability to create posts.'
+        },
+        disabled_members_due_to_create_posts: {
+            id: 'admin.channel_settings.channel_moderation.channelMentions.disabledMemberDueToCreatePosts',
+            defaultMessage: 'Members can not use channel mentions without the ability to create posts.'
+        },
+        disabled_both_due_to_create_posts: {
+            id: 'admin.channel_settings.channel_moderation.channelMentions.disabledBothDueToCreatePosts',
+            defaultMessage: 'Guests and members can not use channel mentions without the ability to create posts.'
+        }
+    },
+    system_scheme: {
+        link: '/admin_console/user_management/permissions/system_scheme',
+        id: 'admin.channel_settings.channel_moderation.systemScheme',
+        defaultMessage: 'System Scheme',
+    },
+    team_scheme: {
+        link: '/admin_console/user_management/permissions/team_override_scheme/',
+        id: 'admin.channel_settings.channel_moderation.teamScheme',
+        defaultMessage: 'Team Scheme',
+    }
+};
+
+interface State {
+    response: Array<ChannelPermissions>;
+}
+
+interface Props {
+    channelPermissions?: Array<ChannelPermissions>;
+    onChannelPermissionsChanged: (name: string, guestsOrMembers: 'guests' | 'members') => void;
+    teamScheme?: Scheme;
+}
+
+interface RowProps {
+    name: string;
+    guests?: any;
+    members?: any;
+    onClick: (name: string, guestsOrMembers: 'guests' | 'members') => void;
+    teamScheme?: Scheme;
+    createPostsObject?: ChannelPermissions | null;
+}
+
+const ChannelModerationTableRow: React.FunctionComponent<RowProps> = (props: RowProps): JSX.Element => {
+    let disabledKey;
+    let isGuestsDisabled = props.guests && !props.guests.enabled;
+    let isMembersDisabled = props.members && !props.members.enabled;
+    if (isGuestsDisabled && isMembersDisabled) {
+        disabledKey = 'disabled_both';
+    } else if (isGuestsDisabled) {
+        disabledKey = 'disabled_guests';
+    } else if (isMembersDisabled) {
+        disabledKey = 'disabled_members';
+    }
+
+    let disabledLinkKey = '';
+    if (disabledKey && props.teamScheme) {
+        disabledLinkKey = 'team_scheme';
+    } else if (disabledKey) {
+        disabledLinkKey = 'system_scheme';
+    }
+
+    let disabledDueToCreatePosts;
+    if (props.createPostsObject) {
+        if (!props.createPostsObject.roles.guests!.value && !props.createPostsObject.roles.members!.value) {
+            disabledDueToCreatePosts = 'disabled_both_due_to_create_posts';
+            disabledKey = '';
+            isGuestsDisabled = true;
+            isMembersDisabled = true;
+            if (props.members.value === true) {
```

Comment:
`members` and `guests` are both optional. Will accessing the `value` key of either (like here) throw an error if null? Should the prop be required or should this use [optional chaining](https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining) (`props.member?.value`) perhaps?